src/databricks_ai_bridge/genie.py

```diff
@@ -213,45 +214,136 @@ def poll_query_results(
-                if resp["status"] == "COMPLETED":
-                    attachment = next((r for r in resp["attachments"] if "query" in r), None)
-                    if attachment:
-                        query_obj = attachment["query"]
-                        description = query_obj.get("description", "")
-                        query_str = query_obj.get("query", "")
-                        attachment_id = attachment["attachment_id"]
-                        return poll_query_results(
-                            attachment_id, query_str, description, returned_conversation_id
-                        )
-                    if resp["status"] == "COMPLETED":
-                        text_content = next(r for r in resp["attachments"] if "text" in r)["text"][
-                            "content"
-                        ]
+
+            # use MLflow client to get parent of any new spans we create from the current active span
+            # (parenting keeps spans in the same trace)
+            client = mlflow.tracking.MlflowClient()
+            with mlflow.start_span(name="genie_timeline", span_type="CHAIN") as parent:
+                parent_trace_id = parent.trace_id if parent else None
+                parent_span_id = parent.span_id if parent else None
+
+                # Track last status from API and the current child span
+                last_status = None
+                current_span = None
+
+                while iteration_count < MAX_ITERATIONS:
+                    iteration_count += 1
+                    resp = self.genie._api.do(
+                        "GET",
+                        f"/api/2.0/genie/spaces/{self.space_id}/conversations/{conversation_id}/messages/{message_id}",
+                        headers=self.headers,
+                    )
+                    returned_conversation_id = resp.get("conversation_id", None)
+
+                    # get current status from API response
+                    current_status = resp["status"]
+
+                    # On status change: end previous span, start a new one (excluding terminal states)
+                    if current_status != last_status:
+                        # END previous span
+                        if current_span is not None:
+                            client.end_span(
+                                trace_id=parent_trace_id,
+                                span_id=current_span.span_id,
+                                attributes={"final_state": last_status},
+                            )
+                            current_span = None
+
+                        # START new span for non-terminal states
+                        if current_status not in {
+                            "COMPLETED",
+                            "FAILED",
+                            "CANCELLED",
+                            "QUERY_RESULT_EXPIRED",
+                        }:
+                            current_span = client.start_span(
```

Comment:
can we define the terminal statuses at the module-level where we defined our other constants so 
`TERMINAL_STATES = {"COMPLETED", "FAILED", "CANCELLED", "QUERY_RESULT_EXPIRED"}` and it's clear what these statuses are for, then we can simply do `if current_status not in TERMINAL_STATES` for better readability :)